src/main/resources/com/google/api/codegen/go/main.snip

```diff
@@ -1,397 +1,248 @@
-                                })
-                            }),
-                        },
-                    @end
-                @end
+            @join def : view.retryPairDefinitions
+              {"{@def.name.params}", "{@def.name.codes}"}: {
+                  gax.WithRetry(func() gax.Retryer {
+                      return gax.OnCodes([]codes.Code{
+                          @join code : def.retryCodes
+                              codes.{@code},
+                          @end
+                      }, gax.Backoff{
+                          Initial: {@def.params.getInitialRetryDelay.getMillis}*time.Millisecond,
+                          Max: {@def.params.getMaxRetryDelay.getMillis}*time.Millisecond,
+                          Multiplier: {@def.params.getRetryDelayMultiplier},
+                      })
+                  }),
+              },
             @end
         }
-
-        return &{@context.getClientPrefix(service)}CallOptions{
-            @join method : context.getNonStreamingMethods(service)
-                @let methodName = method.getSimpleName, \
-                     methodConfig = context.getApiConfig.getInterfaceConfig(service).getMethodConfig(method), \
-                     retryParamsName = methodConfig.getRetrySettingsConfigName, \
-                     retryCodesName = methodConfig.getRetryCodesConfigName
-                    {@methodName}: retry[[2]string{"{@retryParamsName}", "{@retryCodesName}"}],
-                @end
+        return &{@view.callOptionsTypeName}{
+            @join settings : view.callSettings
+                {@settings.methodName}: retry[[2]string{"{@settings.retryParamsName}", "{@settings.retryCodesName}"}],
             @end
         }
     }
-@end
-
-@private constructor(service)
-    @let clientName = context.getClientName(service)
-        // New{@clientName} creates a new {@context.getReducedServiceName(service)} service client.
-        //
-        @join commentLine : context.comments(service)
-            {@commentLine}
-        @end
-        func New{@clientName}(ctx context.Context, opts ...option.ClientOption) (*{@clientName}, error) {
-            conn, err := transport.DialGRPC(ctx, append(default{@clientName}Options(), opts...)...)
-            if err != nil {
-                return nil, err
-            }
-            c := &{@clientName} {
-                conn: conn,
-                client: {@context.getServiceClientConstructorName(service)}(conn),
-                CallOptions: default{@context.getClientPrefix(service)}CallOptions(),
-            }
-            c.SetGoogleClientInfo("gax", gax.Version)
-            return c, nil
-        }
 
-        // Connection returns the client's connection to the API service.
-        func (c *{@clientName}) Connection() *grpc.ClientConn {
-            return c.conn
-        }
+    // {@view.clientName} is a client for interacting with {@view.serviceOriginalName}.
```

Comment:
if serviceOriginalName is only used in docs, I suggest using model.getServiceConfig().getTitle() instead, and remove getServiceOriginalName() from SurfaceNamer.